airflow/utils/types.py

```diff
@@ -22,3 +22,13 @@ class DagRunType(enum.Enum):
     BACKFILL_JOB = "backfill"
     SCHEDULED = "scheduled"
     MANUAL = "manual"
+
+    @staticmethod
+    def resolve_run_type(run_id: str) -> "DagRunType":
+        """
+        Resolved DagRun type from run_id.
+        """
+        for run_type in DagRunType:
+            if run_id.startswith(run_type.value):
```

Comment:
```suggestion
            if run_id.startswith(run_type.value + "__"):
```

no?

Reply:
You marked this as resolved but didn't give any justification.

Reply:
I thought I fiexd it

Reply:
Done now